### PR TITLE
r/aws_redshift_snapshot_copy: new resource

### LINKED ISSUE #36637: [New Resource]: `aws_redshift_snapshot_copy`
### Description

This resource would allow for management of a Redshift cluster's snapshot copy configuration in a standalone resource. Once available, the corresponding `snapshot_copy` argument in the Redshift cluster resource can be deprecated and eventually removed. 

A standalone resource would allow logic and behavior of the snapshot copy configuration lifecycle to be isolated, rather than merged in with cluster logic, preventing errors like #36636.

### Requested Resource(s) and/or Data Source(s)

- `aws_redshift_snapshot_copy`

### Potential Terraform Configuration

```terraform
resource "aws_redshift_snapshot_copy" "example" {
  cluster_id                       = aws_redshift_cluster.example.id
  destination_region               = ""
  manual_snapshot_retention_period = 7
  retention_period                 = 7
  snapshot_copy_grant_name         = ""
}
```

### References

- https://docs.aws.amazon.com/redshift/latest/APIReference/API_EnableSnapshotCopy.html
- https://docs.aws.amazon.com/redshift/latest/APIReference/API_DisableSnapshotCopy.html
- https://docs.aws.amazon.com/redshift/latest/APIReference/API_ModifySnapshotCopyRetentionPeriod.html

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by jar-b**
### Description

This resource will allow practitioners to manage Amazon Redshift cluster snapshot copy configurations with Terraform. This standalone resource will replace the optional `snapshot_copy` block within the `aws_redshift_cluster` resource, allowing snapshot copy configurations to be managed independent of the parent cluster.

The `snapshot_copy` argument within the `aws_redshift_cluster` resource is now deprecated.

### Relations

Closes #36637

### References

- https://docs.aws.amazon.com/redshift/latest/APIReference/API_EnableSnapshotCopy.html
- https://docs.aws.amazon.com/redshift/latest/APIReference/API_DisableSnapshotCopy.html
- https://docs.aws.amazon.com/redshift/latest/APIReference/API_ModifySnapshotCopyRetentionPeriod.html

### Output from Acceptance Testing

```console
% make testacc PKG=redshift TESTS=TestAccRedshiftSnapshotCopy_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.21.8 test ./internal/service/redshift/... -v -count 1 -parallel 20 -run='TestAccRedshiftSnapshotCopy_'  -timeout 360m

--- PASS: TestAccRedshiftSnapshotCopy_disappears_Cluster (411.83s)
--- PASS: TestAccRedshiftSnapshotCopy_disappears (420.58s)
--- PASS: TestAccRedshiftSnapshotCopy_retentionPeriod (422.03s)
--- PASS: TestAccRedshiftSnapshotCopy_basic (423.97s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/redshift   429.531s
```

### CODE REVIEW

File: internal/service/redshift/cluster.go [hunk 1]
```diff
@@ -363,7 +363,9 @@ func ResourceCluster() *schema.Resource {
 				ForceNew: true,
 			},
 			"snapshot_copy": {
-				Type:     schema.TypeList,
+				Type: schema.TypeList,
+				Deprecated: "Use the aws_redshift_snapshot_copy resource instead. " +
+					"This argument will be removed in a future major version.",
```
Comment (ewbankkit) [inline internal/service/redshift/cluster.go:368]: Should this now also be `Computed: true`?

Comment (jar-b) [reply internal/service/redshift/cluster.go:368]: Yes, thanks!

Comment (jar-b) [reply internal/service/redshift/cluster.go:368]: Thanks for pointing this out - I've made the argument optional/computed and removed all the bits about `ignore_changes` from the docs and acceptance tests.

**Comment by jar-b** [comment]
Re-running acceptance tests after making the cluster `snapshot_copy` argument computed and removing `lifecycle.ignore_changes`:

```console
% make testacc PKG=redshift TESTS=TestAccRedshiftSnapshotCopy_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.21.8 test ./internal/service/redshift/... -v -count 1 -parallel 20 -run='TestAccRedshiftSnapshotCopy_'  -timeout 360m
=== RUN   TestAccRedshiftSnapshotCopy_basic
=== PAUSE TestAccRedshiftSnapshotCopy_basic
=== RUN   TestAccRedshiftSnapshotCopy_disappears
=== PAUSE TestAccRedshiftSnapshotCopy_disappears
=== RUN   TestAccRedshiftSnapshotCopy_disappears_Cluster
=== PAUSE TestAccRedshiftSnapshotCopy_disappears_Cluster
=== RUN   TestAccRedshiftSnapshotCopy_retentionPeriod
=== PAUSE TestAccRedshiftSnapshotCopy_retentionPeriod
=== CONT  TestAccRedshiftSnapshotCopy_basic
=== CONT  TestAccRedshiftSnapshotCopy_disappears_Cluster
=== CONT  TestAccRedshiftSnapshotCopy_disappears
=== CONT  TestAccRedshiftSnapshotCopy_retentionPeriod
--- PASS: TestAccRedshiftSnapshotCopy_disappears_Cluster (402.67s)
--- PASS: TestAccRedshiftSnapshotCopy_disappears (410.17s)
--- PASS: TestAccRedshiftSnapshotCopy_basic (412.49s)
--- PASS: TestAccRedshiftSnapshotCopy_retentionPeriod (423.15s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/redshift   428.643s
```

**Pull request merged**